### PR TITLE
shapeit: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package shapeit

### CODE REVIEW

File: var/spack/repos/builtin/packages/shapeit/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    """SHAPEIT is a fast and accurate method for estimation of haplotypes (aka
+       phasing) from genotype or sequencing data."""
+
+    homepage = "https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html"
+    url      = "https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.v2.r837.GLIBCv2.12.Linux.dynamic.tgz"
+
+    version('r837', '895873bb655a0a985cbfd870fdd1dd60')
+
+    def url_for_version(self, version):
+        url = 'https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.v2.{0}.GLIBCv2.12.Linux.dynamic.tgz'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/shapeit/package.py]: I'm sorry that you keep running into all of these weird URLs. Our version support will definitely never be good enough to figure out what the version is in this mess of a URL.

File: var/spack/repos/builtin/packages/shapeit/package.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class Shapeit(Package):
+    """SHAPEIT is a fast and accurate method for estimation of haplotypes (aka
+       phasing) from genotype or sequencing data."""
+
+    homepage = "https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html"
+    url      = "https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.v2.r837.GLIBCv2.12.Linux.dynamic.tgz"
+
+    version('r837', '895873bb655a0a985cbfd870fdd1dd60')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/shapeit/package.py]: Do we want the version to be `r837` or `837`? How do the developers refer to it?

Comment (akthoma) [inline var/spack/repos/builtin/packages/shapeit/package.py]: On the website, the developers say "The latest software release is v2 (r837) ."

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/shapeit/package.py]: Hmm, I vote for calling it `2.837`. You can access the first and second elements with `version[0]` and `version[1]`.

**Pull request merged**